feat: categories for cloudwatch log entries

ml019 opened the pull request:
## Intent of Change
- New feature (non-breaking change which adds functionality)

## Description
Support the ability for a cloud watch log event to explicitly include a unique list of categories that it should be saved under.

If no explicit categories are provided, the event is always considered to carry the "Logs" category.

If multiple categories, return a record per category.

Provide the category as a partition value.

## Motivation and Context
Support separation of logs into categories to reduce the volume of records to be processed by specific consumers such as the SIEM.

## How Has This Been Tested?
Customer deployment

## Related Changes
<!--- If anything not covered by the headings below, add here  -->

### Prerequisite PRs:
<!--- Add a checklist of items or leave the default of "None" -->
- None

### Dependent PRs:
<!--- Add a checklist of items or leave the default of "None" -->
- None

### Consumer Actions:
<!--- Add a checklist of items or leave the default of "None"
What changes must a consumer of this repository make in order to utilise it?
-->
- None

ml019 committed:
feat: categories for cloudwatch log entries

Support the ability for a cloud watch log event to explicitly include a
unique list of categories that it should be saved under.

If no explicit categories are provided, the event is always considered
to carry the "Logs" category.

If multiple categories, return a record per category.

Provide the category as a partition value.

ml019 commented:
@roleyfoley very much a newbie to python so feel free to correct/advise as you see fit.

roleyfoley commented on cloudwatch-firehose-es/src/run.py:
```diff
@@ -111,11 +111,46 @@ def processRecords(records):
-                        'result' : 'Ok',
-                        'recordId' : recId
-                    }
+                    # Determine if extra ways to categorize the event were provided
```
It might be safer here to add this as an optional feature that can be enabled with an env var 

So something like 

```
if os.environ.get("LOG_CATEGORIES", None) is not None: 
```

and at the top just need to import os 
```
import os 
```

ml019 committed:
refactor: add selective enabling of categorization

Add an environment variable to enable the functionality to categorize
log messages.
